src/esm_runscripts/coupler.py

```diff
@@ -11,23 +11,23 @@ class coupler_class:
     def __init__(self, full_config, name):
         self.name = name
 
-        self.process_ordering = full_config[name]["process_ordering"]
-        self.coupled_execs = []
-        for exe in self.process_ordering:
-            self.coupled_execs.append(full_config[exe]["executable"])
-        self.runtime = full_config["general"]["runtime"][5]
-        self.nb_of_couplings = 0
-        if "coupling_target_fields" in full_config[self.name]:
-            for restart_file in list(full_config[self.name]["coupling_target_fields"]):
-                self.nb_of_couplings += len(
-                    list(full_config[self.name]["coupling_target_fields"][restart_file])
-                )
-        if "coupling_input_fields" in full_config[self.name]:
-            for restart_file in list(full_config[self.name]["coupling_input_fields"]):
-                self.nb_of_couplings += len(
-                    list(full_config[self.name]["coupling_input_fields"])
-                )
         if name == "oasis3mct":
+            self.process_ordering = full_config[name]["process_ordering"]
+            self.coupled_execs = []
+            for exe in self.process_ordering:
+                self.coupled_execs.append(full_config[exe]["executable"])
+            self.runtime = full_config["general"]["runtime"][5]
+            self.nb_of_couplings = 0
+            if "coupling_target_fields" in full_config[self.name]:
+                for restart_file in list(full_config[self.name]["coupling_target_fields"]):
+                    self.nb_of_couplings += len(
+                        list(full_config[self.name]["coupling_target_fields"][restart_file])
+                    )
+            if "coupling_input_fields" in full_config[self.name]:
+                for restart_file in list(full_config[self.name]["coupling_input_fields"]):
+                    self.nb_of_couplings += len(
+                        list(full_config[self.name]["coupling_input_fields"])
+                    )
```

Comment:
TODO for @mandresm: if this belongs exclusively to oasis, move it to the `oasis.py`